brew-rmtree.rb

```diff
@@ -29,61 +29,67 @@ def bash(command)
     return %x! bash -c #{escaped_command} !
   end
 
-  def rmtree
-    if ARGV.size < 1 or ['-h', '?', '--help'].include? ARGV.first
-      puts USAGE
-      exit 0
-    end
-
-    raise KegUnspecifiedError if ARGV.named.empty?
-
-    if not ARGV.force?
-      ARGV.named.each do |keg_name|
+  def remove_keg(keg_name)
+    # Remove old versions of keg
+    puts bash "brew cleanup #{keg_name}"
 
-        # Remove old versions of keg
-        puts bash "brew cleanup #{keg_name}"
-
-        # Remove current keg
-        puts bash "brew uninstall #{keg_name}"
-
-        deps = bash "join <(brew leaves) <(brew deps #{keg_name})"
-
-        if deps.length > 0
-          puts "Found lingering dependencies"
-          puts deps.chomp
+    # Remove current keg
+    puts bash "brew uninstall #{keg_name}"
+  end
 
-          deps = deps.split("\n")
+  def deps(keg_name)
+    deps = bash "join <(brew leaves) <(brew deps #{keg_name})"
+    deps.split("\n")
+  end
 
-          deps.each do |dep|
-            dep = dep.chomp
+  def reverse_deps(keg_name)
+    reverse_deps = bash "brew uses --installed #{keg_name}"
+    reverse_deps.split("\n")
+  end
 
-            if !dep.empty?
-              # Check if anything currently installed uses the dependency
-              dep_deps = bash "brew uses --installed #{dep}"
-              dep_deps = dep_deps.chomp
+  def rmtree(keg_name)
+    # Check if anything currently installed uses the keg
+    reverse_deps = reverse_deps(keg_name)
 
-              if dep_deps.length > 0
-                puts "Not removing dependency #{dep} because other installed packages depend on it:"
-                puts dep_deps
-              else
-                # Nothing claims to depend on it
-                puts "Removing dependency #{dep}..."
-                puts bash "brew rmtree #{dep}"
-              end
-            end
-          end
-        else
-          puts "No dependencies left on system for #{keg_name}."
+    if reverse_deps.length > 0
+      puts "Not removing #{keg_name} because other installed kegs depend on it:"
```

Comment:
Correct me if I'm mistaken, but this seems to change the core behavior. The original version will always remove the keg you want removed. What is questioned is whether or not its dependencies can be safely removed. In the new version, the keg you want removed may not be removed if something depends on it. 

I think this is arguable as to what should be the default behavior. While yours is "safer", I was aiming for something you could use in place of your normal uninstall command. But instead of doing that + more, it now does something different.

Was this change in design intentional and you'd like to discuss it, or was this unintentional?